### PR TITLE
update transducer description

### PR DESCRIPTION
**Pull request by cyrilleponcelet**
linked to issue #12. 
update transducer description to be more generic and take into account for Rx and Tx types

### CODE REVIEW

**Comment by gavinmacaulay** [comment]
The changes to the /Platform group look clear. 

The beam_rx_transducer_index variable in Beam_groupX allows for the receiving transducer to be different for each ping. Does this happen in practice?

I have a few minor edits to the grammar - suggestions on how best for me to make these? A pull request on your repository clone, or a list of changes to you?

**Comment by cyrilleponcelet** [comment]
The beam_rx_transducer_index variable in Beam_groupX allows for the receiving transducer to be different for each ping. Does this happen in practice?
=> After talks with my colleagues this does not happen in practice for rx_beam (but it happens for tx). As far as we know when this happens it is linked to a change in configuration and number of beams. In such a case the sounder cut the file and open a new one. I think I did it to be coherent with tx and take into account a use case when I'm merging several files. We can remove the ping dimension if you wish.

For the second point I'm not familiar with github. Are you able to modify directly in my branch ? I can recreate a new pull request with you modification
Have you tried the review function of github ? Never used it but it seems the tool to use

File: SONAR-netCDF4/docs/tablePlatform.adoc [hunk 1]
```diff
@@ -1,10 +1,15 @@
-|==============================================================================================================================================================================================================
+﻿|==============================================================================================================================================================================================================
 |Description |Obligation |Comment
+
 |Group attributes | |
 |:platform_code_ICES |O |Platform code from the ICES SHIPC vocabulary (http://vocab.ices.dk/services/pox/GetCodeList/SHIPC).
 |:platform_name |O |Platform name of which the sonar is a part.
 |:platform_type |O |Platform type that the sonar is part of. Use the description field from the ICES ”Platform Class” vocabulary (http://vocab.ices.dk/services/pox/GetCodeList/Platform%20Class).
+|Types | |
+|byte enum transducer_type_t {rx_transducer = 0, tx_transducer = 1, monostatic_transducer = 3 } | |transducer type indicate if the transducer is a Tx Transducer, Rx Transducer or both (monostatic)
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tablePlatform.adoc]:
```suggestion
|byte enum transducer_type_t {receive_only = 0, transmit_only = 1, monostatic = 3 } | |Transducer function - transmit only, receive only or both (monostatic)
```

File: SONAR-netCDF4/docs/tablePlatform.adoc [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+
 |Dimensions | |
+|transducer:transducer |M |transducer count
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tablePlatform.adoc]:
```suggestion
|transducer:transducer |M |Transducer count
```

File: SONAR-netCDF4/docs/tablePlatform.adoc [hunk 3]
```diff
@@ -152,27 +157,44 @@ a|
...
-|float transducer_offset_z |R |Distance from the platform coordinate system origin to the monostatic sonar transducer along the _z_-axis.
-|:long_name =  +
-"_z_-axis distance from the +
+|float transducer_offset_z(transducer) |R |Distance from the platform coordinate system origin to the transducer along the _z_-axis.
+|:long_name = "_z_-axis distance from the +
 platform coordinate system +
 origin to the sonar transducer" | |
 |:units = "m" | |
 | | |
+|int transducer(transducer) |R |transducer serial number
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tablePlatform.adoc]: Not all transducers have a serial number that is a number (some have letters). Suggest to change this to:

```suggestion
string| transducerId(transducer) |R |Transducer serial number or identification name
```

File: SONAR-netCDF4/docs/tablePlatform.adoc [hunk 4]
```diff
@@ -152,27 +157,44 @@ a|
...
+|float :valid_range = -180.0f, 180.0f | |
+|:units = "arc_degree" | |
+|:long_name = "Extrinsic rotation about the _y_-axis from the transducer to reference coordinate systems" | |
+| | |
+|float transducer_rotation_z(transducer) |R |Extrinsic angular rotation about the _z_-axis from the transducer zero angle to the coordinate system origin zero angle.
+|float :valid_range = -180.0f, 180.0f | |
+|:units = "arc_degree" | |
+|:long_name = "Extrinsic rotation about the _z_-azis from the transducer to reference coordinate systems" | |
+| | |
+|transducer_type_t transducer_type(transducer) |M |The type of transducer, might be rx , tx or monostatic
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tablePlatform.adoc]: Perhaps change to transducer_function (or something else that suggests the transmit only, receive only, monostatic thing). Transducer type can be a quite general term (e.g, Tonplitz, composite,  broadband, etc)
```suggestion
|transducer_type_t transducer_function(transducer) |M |The transducer function (that is, transmit_only, receive_only, or monostatic)
```

File: SONAR-netCDF4/docs/tablePlatform.adoc [hunk 5]
```diff
@@ -152,27 +157,44 @@ a|
...
+|:long_name = " type of transducer" | |
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tablePlatform.adoc]:
```suggestion
|:long_name = "Transducer function (transmit_only, receive_only, monostatic)" | |
```

File: SONAR-netCDF4/docs/tableBeamGroup1.adoc [hunk 6]
```diff
@@ -158,4 +158,8 @@ transmitted pulse" | |
 | | |
 |transmit_t transmit_type(ping_time) |M |Type of transmit pulse.
 |:long_name = "Type of transmitted pulse" | |
+| | |
+|int beam_rx_transducer_index(ping_time, beam) |MA |receiving or monostatic transducer index associated with the given beam
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tableBeamGroup1.adoc]:
```suggestion
|int receive_transducer_index(beam) |MA |Receiving or monostatic transducer index associated with the given beam
```

File: SONAR-netCDF4/docs/tableBeamGroup1.adoc [hunk 7]
```diff
@@ -158,4 +158,8 @@ transmitted pulse" | |
...
+|:valid_min = "0" | |
+|:long_name = "receiving transducer index associated with the given beam" | |
```
Comment (gavinmacaulay) [inline SONAR-netCDF4/docs/tableBeamGroup1.adoc]: Tried to be have a clearer long name. Not sure if I succeeded...
```suggestion
|:long_name = "Index into the transducer vector variables to indicate transducer function" | |
```

**Comment by gavinmacaulay** [comment]
Apart from the small changes and suggestions above, I think that this pull request is fine.

File: SONAR-netCDF4/docs/tablePlatform.adoc (hunk 3 above)
Comment (cyrilleponcelet) [reply SONAR-netCDF4/docs/tablePlatform.adoc]:
```suggestion
|string transducerId(transducer) |R |Transducer serial number or identification name
```
Just a pip misplaced

File: SONAR-netCDF4/docs/tableBeamGroup1.adoc (hunk 7 above)
Comment (cyrilleponcelet) [reply SONAR-netCDF4/docs/tableBeamGroup1.adoc]: Not convinced by this one, it does not indicate the transducer function (we can have several transducer with the same function) but the transducer itself.

Comment (cyrilleponcelet) [reply SONAR-netCDF4/docs/tableBeamGroup1.adoc]: Not sure either, everytime it is paraphrasing the comment of the variable ...

```suggestion
|:long_name = "Index into the transducer vector variables linking a beam to the transducer" | |
```

Comment (gavinmacaulay) [reply SONAR-netCDF4/docs/tableBeamGroup1.adoc]: Good point about the paraphrasing of the comment. The netCDF/CF guidelines for long_name is something that is useful for a plot label. With that guidance, how about simply "Receive transducer index"?

Comment (cyrilleponcelet) [reply SONAR-netCDF4/docs/tableBeamGroup1.adoc]: Good for me, we use long names in our applications to display axis selection and if it is too long it's become an issue for us.
I will commit your suggestion

```suggestion
|:long_name = "Receive transducer index" | |
```

**Comment by cyrilleponcelet** [comment]
OK Gavin I do not know if I did it right but it seems ready to merge

**Pull request merged**